main.js

```diff
@@ -53,56 +172,118 @@ function longestWord(sen) {
+      temp = []
+    }
+  }
+  return ans
+}
 
 
 
 /* 9. Take an array of arrays and flatten to a single array
  * ex. flattenArray([[1, 2], [3, 4], [5, 6], [7]]) should return [1, 2, 3, 4, 5, 6, 7]
  */
-function flattenArray(arrays) {}
-
-
+function flattenArray(arrays) {
+  return [].concat(...arrays)
+}
 
 /* 10. Return true if anagram and false if not
  * ex. isAnagram('elbow', 'below') === true
  */
-function isAnagram(str1, str2) {}
+function isAnagram(str1, str2) {
+  return str1.split("").sort().join("") === str2.split("").sort().join("")
+}
 
 
 
 /* 11. Change every letter of the string to the one that follows it (circularly), and capitalize the vowels
  * ex. 'hello there' === 'Ifmmp UIfsf'
  */
-function letterChanges(str) {}
+function letterChanges(str) {
+  let ans = ""
+  str.split("").forEach(x => {
+
+    let c = x.charCodeAt(0)
+    let temp = ""
+    if (c == 90) {temp = 'A'}
+    else if (c == 122) {temp = 'a'}
+    else if (c == 32) {temp = ' '}
+    else {temp = String.fromCharCode(++c)}
+    if (/[aeiou]/.test(temp)) {temp = temp.toUpperCase()}
+    ans += temp
+  });
+  return ans
+}
 
 
 
 /* 12. Return a sum of all parameters entered regardless of the amount of numbers - NO ARRAYS
  * ex. addAll(2,5,6,7) === 20
  */
-function addAll() {}
+function addAll(...arr) {
+  let sum = 0
+  arr.forEach(x => {
+    sum += x
+  });
+  return sum
+}
 
 
 
 /* 13. Pass in a number to loop up to and add all of the prime numbers. A prime number is a whole number greater than 1 whose only factors are 1 and itself
  * ex. sumAllPrimes(10) == 17
  */
-function sumAllPrimes() {}
+function sumAllPrimes(end) {
+  let sum = 0
+  function isPrime(num) {
+    for(let i = 2, s = Math.sqrt(num); i <= s; i++) {
+        if(num % i === 0) return false;
+    }
+    return num > 1;
+  }
+  for (let x = 2; x <= end; x++) {if (isPrime(x)) {sum += x}}
+  return sum
+}
 
 
 
 /* 14. Remove from the array whatever is in the following arguments. Return the leftover numbers in an array
  * ex. seekAndDestroy([2, 3, 4, 6, 6, 'hello'], 2, 6) == [3, 4, 'hello']
  */
-function seekAndDestroy() {}
+function seekAndDestroy(arr,...rem) {
+  let ans = []
+  arr.forEach(element => {
+    if (!rem.includes(element)) {ans.push(element)}
+  });
+  return ans
+}
 
 
 
-/* 15. Some people are standing in a row in a park. There are trees between them which cannot be moved. Your task is to rearrange the people by their heights in a non-descending order without moving the trees.
+/* 15. Some people are standing in a row in a park. There are trees between them which cannot be moved.
+       Your task is to rearrange the people by their heights in a non-descending order without moving the trees.
  * ex. sortByHeight([-1, 150, 190, 170, -1, -1, 160, 180]) == [-1, 150, 160, 170, -1, -1, 180, 190]
  */
-function sortByHeight() {}
+function sortByHeight(arr) {
+  let res = []
+  let i = 0
+  arr.forEach(element => { if (element != -1) {res.push(element)}})
+  res = res.sort()
```

Comment:
array sort makes in-place changes, it doesn't return a sorted copy of the original. it is a mutating function that returns void